src/main/java/edu/harvard/iq/dataverse/api/Files.java

```diff
@@ -392,16 +392,32 @@ public Response updateFileMetadata(@FormDataParam("jsonData") String jsonData,
                 javax.json.JsonObject jsonObject = jsonReader.readObject();
                 String label = jsonObject.getString("label", null);
                 String directoryLabel = jsonObject.getString("directoryLabel", null);
-                String path = "";
-                if (directoryLabel != null) {
-                    path = directoryLabel + "/";
+                // If the user is trying to change the label/directoryLabel or not.
+                boolean labelChange = true;
+                if (label == null && directoryLabel == null) {
+                    labelChange = false;
+                }
+                String oldLabel = df.getFileMetadata().getLabel();
+                String oldDirectoryLabel = df.getFileMetadata().getDirectoryLabel();
+                String oldPathPlusName = oldDirectoryLabel + "/" + oldLabel;
+                if (directoryLabel == null) {
+                    directoryLabel = oldDirectoryLabel;
                 }
                 if (label == null) {
-                    label = df.getFileMetadata().getLabel();
+                    label = oldLabel;
+                }
+                String incomingPathPlusName = directoryLabel + "/" + label;
+                if (oldPathPlusName.equals(incomingPathPlusName)) {
+                    labelChange = false;
                 }
-                if (IngestUtil.conflictsWithExistingFilenames(label, directoryLabel, fmdList, df)) {
-                    String incomingPathPlusFileName = path + label;
-                    return error(BAD_REQUEST, BundleUtil.getStringFromBundle("files.api.metadata.update.duplicateFile", Arrays.asList(incomingPathPlusFileName)));
+                if (labelChange && IngestUtil.conflictsWithExistingFilenames(label, directoryLabel, fmdList, df)) {
+                    String pathPlusFilename = "";
```

Comment:
Because if we do (I just tried), we get "null/" as part of the output. Tests fail with messages like this:

```
Expected: Filename already exists at README.md
  Actual: Filename already exists at null/README.md
```

That is to say, that code (below) is there to make sure there's no "null/" in the output, like above.

    String pathPlusFilename = "";
    if (directoryLabel != null) {
        pathPlusFilename = directoryLabel + "/" + label;
    } else {
        pathPlusFilename = label;
    }